ebos/eclmpiserializer.hh

```diff
@@ -145,6 +209,18 @@ protected:
             const_cast<T2&>(data.second).serializeOp(*this);
     }
 
+    template<class T1>
+    void shared_ptr(const std::shared_ptr<T1>& data)
+    {
+        bool value = data ? true : false;
+        (*this)(value);
+        if (m_op == Operation::UNPACK && value) {
+            const_cast<std::shared_ptr<T1>&>(data) = std::make_shared<T1>();
```

Comment:
Does this require a default constructor?

Reply:
Yes.